packages/ability/src/ability.ts

```diff
@@ -77,8 +77,8 @@ export function defineRulesForPurchases(purchases: any[]) {
       productId: {
         $in: purchases?.map(
           (purchase: any) =>
-            (purchase.productId && purchase.status === 'Valid') ||
-            purchase.status === 'Restricted',
+            purchase.productId &&
+            ['Valid', 'Restricted'].includes(purchase.status),
```

Comment:
The parentheses were throwing off the precedence of this check. I thought it would be simpler to do an `includes` check for the `purchase.status` instead.